src/renderer/component/publishForm/view.jsx

```diff
@@ -137,13 +139,16 @@ class PublishForm extends React.PureComponent<Props> {
   }
 
   handleChannelChange(channelName: string) {
-    const { name, updatePublishForm } = this.props;
+    const { name, updatePublishForm, myChannels } = this.props;
+    const form = { channel: channelName };
+    const namedChannelClaim = myChannels.filter(channel => channel.name === channelName)[0];
+    if (namedChannelClaim) {
+      form.channelId = namedChannelClaim.claim_id;
```

Comment:
This should clear out the channel id if there is no `namedChannelClaim`

If I select a channel, then go back to anonymous, the same `claim_id` will be stored in redux